src/CommunityToolkit.Maui/Behaviors/ICommunityToolkitBehavior.shared.cs

```diff
@@ -0,0 +1,96 @@
+using System.ComponentModel;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace CommunityToolkit.Maui.Behaviors;
+
+/// <summary>
+/// A common interface to be used across <see cref="BaseBehavior{TView}"/> and <see cref="BasePlatformBehavior{TView,TPlatformView}"/>
+/// </summary>
+public interface ICommunityToolkitBehavior<TView> where TView : Element
```

Comment:
This Interface primarily contains default methods in order to align the functionality of `BaseBehavior` and `BasePlatformBehavior`.

We had to use an interface with default methods because we cannot use the same `abstract class` for both `BaseBehavor : Behavior` and `BasePlatformBehavior : PlatformBehavior` because these base behavior classes inherit from different .NET MAUI classes.

Reply:
Should the interface be `internal`?

Reply:
I wish. 

It has to be `public` because all of our public Behaviors are implementing it. 

I did make all of its methods + properties internal or protected so that we don't need to support it as a stand-alone external interface.